### ISSUE TITLE
Menu: Warning: Can't call setState (or forceUpdate) on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the

### ISSUE THREAD

**Issue by oste** [jamesmfriedman/rmwc#289]
## RMWC or MDC?

RMWC

### What RMWC Version are you using?

"rmwc": "^1.8.3",

### What browser(s) and React Version is this bug affecting?

react 16.4.2, all browsers

### What build system are you using?

Create React App

### What are the steps to reproduce the bug?

When a component using Menu/MenuAnchor is unmounted the onClose callback is fired. This can lead to the Warning for setting state on an unmounted component in the title. 

To reproduce this you can perform a redirect from a component using the Menu component like this

```
menuSelected() {
    this.setState({toProfile: true});
}

render() {
    if (this.state.toProfile === true) {
        return <Redirect to='/profile' />
    }          
        
    <Menu
        open={this.state.menuOpen}
        onClose={() => this.setState({menuOpen: false})}
        anchorCorner='bottomStart'
        onCancel={() => this.setState({menuOpen: false})}
        onSelected={ evt => this.menuSelected(evt)}>
        {menuItems}
    </Menu>
```

### What is the expected behavior?

onClose would be called before any unmounting can take place. 

### What is the actual behavior?

onClose gets called after onSelected. If onSelected has any unmounting behavior such as a redirect like this the onClose with throw this error.

### Any other information you believe would be useful?

Moving this above the SELECTED_EVENT would probably work https://github.com/jamesmfriedman/rmwc/blob/master/src/Menu/index.js#L96. I would propose removing this callback all together and rely on the cancelled and selected methods provided by MDC. That's what I will be doing moving forward.

**Comment by oste** [comment]
Was noticing a sporadic can't remove class in the core mdc libs even when removing `onClose`. Will see if I can reproduce

**Comment by jamesmfriedman** [comment]
Menus are getting refactored in MDC38 / MDC39 so this might be fixed in with the next RMWC release

**Comment by jamesmfriedman** [comment]
Fixed with 1.9 release
